isis/src/kaguya/apps/kaguyami2isis/kaguyami2isis.cpp

```diff
@@ -0,0 +1,159 @@
+    if(ui.GetBoolean("SETHRSRANGE")) {
+      p.SetHRS(ui.GetDouble("HRSMIN"), ui.GetDouble("HRSMAX"));
+    }
+    if(ui.GetBoolean("SETHISRANGE")) {
+      p.SetHIS(ui.GetDouble("HISMIN"), ui.GetDouble("HISMAX"));
+    }
+    if(ui.GetBoolean("SETLRSRANGE")) {
+      p.SetLRS(ui.GetDouble("LRSMIN"), ui.GetDouble("LRSMAX"));
+    }
+    if(ui.GetBoolean("SETLISRANGE")) {
+      p.SetLIS(ui.GetDouble("LISMIN"), ui.GetDouble("LISMAX"));
+    }
+
+    p.SetOrganization(Isis::ProcessImport::BSQ);
+
+    p.StartProcess();
+
+    // Get the directory where the Kaguya MI translation tables are.
+    QString transDir = "$ISISROOT/appdata/translations/";
+    Pvl inputLabel(inFile.expanded());
+    Pvl *outputLabel = outcube->label();
+    FileName transFile;
+
+    // Translate the Archive group
+    transFile = transDir + "KaguyaMiArchive.trn";
+    PvlToPvlTranslationManager archiveXlater(inputLabel, transFile.expanded());
+    archiveXlater.Auto(*(outputLabel));
+
+    // Translate the Instrument group
+    transFile = transDir + "KaguyaMiInstrument.trn";
+    PvlToPvlTranslationManager instrumentXlater(inputLabel, transFile.expanded());
+    instrumentXlater.Auto(*(outputLabel));
+      //trim trailing z's from the time strings
+    PvlGroup &instGroup(outputLabel->findGroup("Instrument",Pvl::Traverse));
+    QString timeString;
+      //StartTime
+    PvlKeyword &startTimeKeyword=instGroup["StartTime"];
+    timeString = startTimeKeyword[0];
+    startTimeKeyword.setValue( timeString.mid(0,timeString.lastIndexOf("Z")) );
+      //StartTimeRaw
+    PvlKeyword &startTimeRawKeyword=instGroup["StartTimeRaw"];
+    timeString = startTimeRawKeyword[0];
+    startTimeRawKeyword.setValue( timeString.mid(0,timeString.lastIndexOf("Z")) );
+      //StopTime
+    PvlKeyword &stopTimeKeyword=instGroup["StopTime"];
+    timeString = stopTimeKeyword[0];
+    stopTimeKeyword.setValue( timeString.mid(0,timeString.lastIndexOf("Z")) );
+      //StopTimeRaw
+    PvlKeyword &stopTimeRawKeyword=instGroup["StopTimeRaw"];
+    timeString = stopTimeRawKeyword[0];
+    stopTimeRawKeyword.setValue( timeString.mid(0,timeString.lastIndexOf("Z")) );
+
+
+    // Translate the BandBin group
+    transFile = transDir + "KaguyaMiBandBin.trn";
+    PvlToPvlTranslationManager bandBinXlater(inputLabel, transFile.expanded());
+    bandBinXlater.Auto(*(outputLabel));
+
+    //Set up the Kernels group
+    PvlGroup kern("Kernels");
+    QString baseBand = lab.findKeyword("BASE_BAND")[0];
+    if (lab.findKeyword("INSTRUMENT_ID")[0] == "MI-VIS") {
+      if (baseBand.contains("MV")) {
+        kern += PvlKeyword("NaifFrameCode", toString(-13133) + baseBand[2]);
+      }
+      kern += PvlKeyword("NaifCkCode", toString(-131330));
+    }
+    else if (lab.findKeyword("INSTRUMENT_ID")[0] == "MI-NIR") {
+      if (baseBand.contains("MN")) {
+        kern += PvlKeyword("NaifFrameCode", toString(-13134) + baseBand[2]);
+      }
+      kern += PvlKeyword("NaifCkCode", toString(-131340));
+    }
```

Comment:
Significant change made here for how the `NaifFrameCode`  is generated. Changed from 5 part if/else blocks to single if that extracts the value of the baseBand number. For `MI-VIS` and `MI-NIR` the baseBand defines the `NaifFrameCode` where the digit in the baseBand is the last digit in the frame code.

Reply:
Yes, the original tests were passing after this change